trie/proof.go

```diff
@@ -464,125 +464,97 @@ func hasRightElement(node node, key []byte) bool {
 //
 // Except returning the error to indicate the proof is valid or not, the function will
 // also return a flag to indicate whether there exists more accounts/slots in the trie.
-func VerifyRangeProof(rootHash common.Hash, firstKey []byte, lastKey []byte, keys [][]byte, values [][]byte, proof ethdb.KeyValueReader) (ethdb.KeyValueStore, bool, error) {
+func VerifyRangeProof(rootHash common.Hash, firstKey []byte, lastKey []byte, keys [][]byte, values [][]byte, proof ethdb.KeyValueReader) (bool, error) {
```

Comment:
We should add a note about the Verifier not checking proof 'canonicality'. E.g. 

```
// Note: VerifyRangeProof does not verify that the proof is on minimal form. If the input proofs 
// are 'bloated' with neighbour leaves or random data, aside from the 'useful' data, then the 
// proof will still be accepted.

Reply:
Added